translation-service-eTranslation/src/main/java/eu/europeana/api/translation/service/etranslation/ETranslationTranslationService.java

```diff
@@ -0,0 +1,211 @@
+package eu.europeana.api.translation.service.etranslation;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.util.EntityUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.codehaus.jettison.json.JSONArray;
+import org.codehaus.jettison.json.JSONException;
+import org.codehaus.jettison.json.JSONObject;
+import org.springframework.data.redis.listener.ChannelTopic;
+import org.springframework.data.redis.listener.RedisMessageListenerContainer;
+import org.springframework.data.redis.listener.adapter.MessageListenerAdapter;
+import eu.europeana.api.translation.definitions.model.TranslationObj;
+import eu.europeana.api.translation.service.AbstractTranslationService;
+import eu.europeana.api.translation.service.exception.TranslationException;
+
+public class ETranslationTranslationService extends AbstractTranslationService {
+  
+  private static final Logger LOGGER = LogManager.getLogger(ETranslationTranslationService.class);
+
+  private String serviceId;
+  private final String baseUrl;
+  private final String domain;
+  //this is the base url of the translation api (without the request handler (or the controller) endpoint)
+  private final String callbackUrl;
+  private final String credentialUsername;
+  private final String credentialPwd;
+  private final int maxWaitMillisec;
+  private final RedisMessageListenerContainer redisMessageListenerContainer;
+  
+  public ETranslationTranslationService(String baseUrl, String domain, String callbackUrl, int maxWaitMillisec, 
+      String username, String password, RedisMessageListenerContainer redisMessageListenerContainer) {
```

Comment:
better use translationConfig for String baseUrl, String domain, String callbackUrl, int maxWaitMillisec, 
      String username, String password,

Reply:
It is anyway used. Here, in the constructor, we cannot use it, because it would require the dependency of the eTranslation module to the web module where the TranslationConfig is defined.